src/compiler/checker.ts

```diff
@@ -1589,6 +1593,12 @@ namespace ts {
             return type;
         }
 
+        function createPartialTypeFromObjectType(target: Type): Type {
+            const type = <PartialType>createType(TypeFlags.Partial | TypeFlags.ObjectType);

```

Comment:
This definitely isn't right. `TypeFlags.ObjectType` is a mask of several other flags, so you'll now appear to be an instance of all of those kinds.